crates/ruma-identifiers-validation/src/error.rs

```diff
@@ -63,3 +63,18 @@ pub enum MxcUriError {
     #[error("invalid Server Name")]
     ServerNameMalformed,
 }
+
+// Required for MxcUri
+impl PartialOrd for MxcUriError {
+    fn partial_cmp(&self, _: &Self) -> Option<std::cmp::Ordering> {
+        None
+    }
+}
+
+// Required for MxcUri
+impl Ord for MxcUriError {
+    fn cmp(&self, _: &Self) -> std::cmp::Ordering {
+        // Makes this effectively a noop in ordering
+        std::cmp::Ordering::Equal
+    }
+}
```

Comment:
Side note: If this were to be kept in `MxcUri`, the right thing to do would be implementing `PartialOrd` and `Ord` for it manually, ignoring the `slash_idx` field rather than implementing the traits for the error type.